File: FrontEnd - ControleDeVacinacaoBovina/angular-controle-de-vacinacao-bovina/src/app/entities/rebanho.ts
import {Especie} from "./especie";
import {Propriedade} from "./propriedade";

export class Rebanho {
  idRebanho: number = 0;
  quantidadeTotal: number = 0;
  quantidadeVacinada: number = 0;
  idEspecie: number = 0;
  idPropriedade: number = 0;
  especie: Especie = new Especie();
  propriedade: Propriedade = new Propriedade();
}

File: BackEnd - ControleDeVacinacaoBovina/Models/Rebanho.cs
using System.Collections.Generic;

namespace ControleDeVacinacaoBovina.Models
{
    public class Rebanho
    {
        public int IdRebanho { get; set; }
        public int QuantidadeTotal { get; set; }
        public int QuantidadeVacinada { get; set; }
        public int IdEspecie { get; set; }
        public int IdPropriedade { get; set; }
        public Especie Especie { get; set; }
        public Propriedade Propriedade { get; set; }

        public List<RegistroVacinacao> RegistroVacinacoes;
        public List<Venda> Vendas;
    }
}